packages/components/src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,107 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconCheckMark, IconCross } from '@frontify/fondue-icons';
+import { type ChangeEvent, type KeyboardEvent, forwardRef, type ReactNode } from 'react';
+
+import { rootStyles, inputStyles, slotStyles, loadingStatusStyles } from './styles/textInputStyles';
+
+import { cn } from '#/utilities/styleUtilities';
+
+type TextInputProps = {
+    id?: string;
+    children?: ReactNode;
+    defaultValue?: string | number;
+    value?: string | number;
+    /**
+     * Type of the input field
+     * @default text
+     */
+    type?: 'date' | 'email' | 'hidden' | 'number' | 'password' | 'search' | 'tel' | 'text' | 'time' | 'url';
+    placeholder?: string;
+    disabled?: boolean;
+    required?: boolean;
+    spellCheck?: boolean;
+    readOnly?: boolean;
+    autoComplete?: string;
+    maxLength?: number;
+    /**
+     * Status of the input field
+     * @default neutral
+     */
+    status?: 'neutral' | 'success' | 'error' | 'loading';
+    className?: string;
+    onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
+    onBlur?: (event: ChangeEvent<HTMLInputElement>) => void;
+    onFocus?: (event: ChangeEvent<HTMLInputElement>) => void;
+    onKeyDown?: (event: KeyboardEvent<HTMLInputElement>) => void;
+    onKeyUp?: (event: KeyboardEvent<HTMLInputElement>) => void;
+    'data-test-id'?: string;
+    'aria-label'?: string;
+    'aria-labelledby'?: string;
+    'aria-describedby'?: string;
+};
+
+const TextFieldRoot = forwardRef<HTMLInputElement, TextInputProps>(
+    ({ children, className, status = 'neutral', 'data-test-id': dataTestId, ...inputProps }, ref) => {
+        return (
+            <div
+                className={cn(rootStyles, className)}
+                data-disabled={inputProps.disabled}
```

Comment:
why is this needed? couldn't this be determined by has:input:disabled (pseudo code)

If we wanna explicitly define those things, we should use tailwind-variants

Reply:
- div shouldn't have `disabled`
- Side-effect of the wrapper based on the input instead of the other way around
- Easier debugging

Reply:
i thought during our discussion yesterday we decided that for these specifically defined attributes we decided to use tailwindVariants?

I'm abolutely okay with using the css selectors for the things that are there anyways or only possible that way (disabled input or slot name) but for the ones specifically defined, pls use variants or let's discuss again.

Reply:
Fixed, pushed in the wrong branch